### PR TITLE
docs: fixed en version of jd usecase index

### PR DESCRIPTION
**Pull request by michelle-qinqin**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
docs

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: docs/en/use_case/JD_recommendation_en.md [hunk 1]
```diff
@@ -8,7 +7,9 @@ Extracting patterns from historical data to predict the future purchase intentio
 
 Note that: (1) this case is based on the OpenMLDB cluster version for tutorial demonstration; (2) this document uses the pre-compiled docker image. If you want to test it in the OpenMLDB environment compiled and built by yourself, you need to configure and use our [Spark Distribution for Feature Engineering Optimization](https://github.com/4paradigm/spark). Please refer to relevant documents of [compilation](https://openmldb.ai/docs/en/main/deploy/compile.html) (Refer to Chapter: "Spark Distribution Optimized for OpenMLDB") and the [installation and deployment documents](https://openmldb.ai/docs/en/main/deploy/install_deploy.html) (Refer to the section: [Deploy TaskManager](https://openmldb.ai/docs/en/main/deploy/install_deploy.html#deploy-taskmanager)).
 
-## 1.  Preparation and Preliminary Knowledge
+## 3. testete
```
Comment (lumianph) [inline docs/en/use_case/JD_recommendation_en.md:10]: what is this?

File: docs/en/use_case/JD_recommendation_en.md [hunk 2]
```diff
@@ -8,7 +7,9 @@ Extracting patterns from historical data to predict the future purchase intentio
...
+## Preparation and Preliminary Knowledge
+## 4 Preparation and Preliminary Knowledge
```
Comment (lumianph) [inline docs/en/use_case/JD_recommendation_en.md:12]: ## 1. Preparation and Preliminary Knowledge

**Review by lumianph — CHANGES_REQUESTED** [review]

File: docs/en/use_case/JD_recommendation_en.md (hunk 1 above)
Comment (michelle-qinqin) [reply docs/en/use_case/JD_recommendation_en.md:10]: My local compiler is broken

Comment (michelle-qinqin) [reply docs/en/use_case/JD_recommendation_en.md:10]: I was testing why the H2 became H1 format after sphinx compling

**Pull request closed without merging**